nbconvert/exporters/webpdf.py

```diff
@@ -105,13 +107,19 @@ async def main():
             return pdf_data
 
         pool = concurrent.futures.ThreadPoolExecutor()
-        # TODO: when dropping Python 3.6, use
-        # pdf_data = pool.submit(asyncio.run, main()).result()
-        def run_coroutine(coro):
-            loop = asyncio.new_event_loop()
-            asyncio.set_event_loop(loop)
-            return loop.run_until_complete(coro)
-        pdf_data = pool.submit(run_coroutine, main()).result()
+        temp_file = tempfile.NamedTemporaryFile(suffix=".html", delete=False)
+        with temp_file:
+            temp_file.write(html.encode('utf-8'))
```

Comment:
I think just a tempfile.flush() here will do the right thing, and we can remove the explicit os.unlink call. IMO these two are almost equivalent, but putting everything in the `with` block is clearer to understand.

Reply:
As I said in the other conversation. According to the documentation, Chrome cannot open the file inside the `with` block on NT. Windows will not allow other process to access an open temp file, it has to be closed first.

Reply:
Ooh, oops. I totally missed that - am really sorry! GitHub had collapsed that comment as 'stale' so I did not see it!!!

Since it's non-obvious, can you just write a comment explaining why this is the case? I think I'm happy to merge after that - don't think we need a specific test case for this

Reply:
Added comments